build_runner/lib/src/entrypoint/run_script.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:build_runner/build_runner.dart';
+import 'package:build_runner_core/build_runner_core.dart';
+import 'package:io/io.dart';
+import 'package:path/path.dart' as p;
+
+import 'base_command.dart';
+import 'options.dart';
+
+class RunCommand extends BuildRunnerCommand {
+  @override
+  String get name => 'run';
+
+  @override
+  String get description =>
+      'Performs a single build on the specified targets, and executes a Dart script with the given arguments.';
+
+  @override
+  String get invocation =>
+      '${super.invocation.replaceFirst('[arguments]', '[build-arguments]')} '
+      '<executable> [-- [script-arguments]]';
+
+  @override
+  SharedOptions readOptions() {
+    // The default option parser will throw if we pass additional arguments,
+    // because it expects positional arguments to be build directories.
+    //
+    // Instead, pass an empty list, and we'll handle positional arguments ourselves.
+    return new SharedOptions.fromParsedArgs(
```

Comment:
See the [Test command](https://github.com/dart-lang/build/blob/master/build_runner/lib/src/entrypoint/test.dart#L33) implementation of this, we should do something similar here to ensure that we only have a single extra argument before the `--` (the script to run), and then additional extra arguments are after the `--`.

Reply:
Ok, changed it - I've more or less just copied the implementation from the test command at this point.